src/DataLayouts/struct.jl

```diff
@@ -227,6 +227,7 @@ end
     offset,
 ) where {S}
     array[offset + 1] = val
+    return nothing
```

Comment:
we should probably return the container value (similar to `setindex!` instead of `nothing`, but good catch.  I think this is fine for now.

Reply:
So, just `return val` here and in the `else` block above?